solr/core/src/java/org/apache/solr/cloud/ShardLeaderElectionContextBase.java

```diff
@@ -179,17 +181,19 @@ void runLeaderProcess(boolean weAreReplacement, int pauseBeforeStartMs)
           ZkStateReader.STATE_PROP, Replica.State.ACTIVE.toString());
       assert zkController != null;
       assert zkController.getOverseer() != null;
-      zkController.getOverseer().offerStateUpdate(Utils.toJSON(m));
+      DocCollection coll = zkStateReader.getCollection(this.collection);
+      if (coll == null || coll.getStateFormat() < 2 || ZkController.sendToOverseer(coll, id)) {
```

Comment:
Nit: first two checks are run in sendToOverseer, are you checking here to avoid opening a new stackframe?

Reply:
yeah, early termination